packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_screen.dart

```diff
@@ -24,8 +130,112 @@ class DeepLinksScreen extends Screen {
 
   @override
   Widget build(BuildContext context) {
-    return const Center(
-      child: Text('TODO: build deep link validation tool'),
+    return const DeepLinkPage();
+  }
+}
+
+class DeepLinkPage extends StatefulWidget {
+  const DeepLinkPage({super.key});
+
+  @override
+  State<DeepLinkPage> createState() => _DeepLinkPageState();
+}
+
+class _DeepLinkPageState extends State<DeepLinkPage> {
+  String responseBody = 'empty response';
+  int? selectedRowIndex;
+  String searchContent = '';
+  bool bundleByDomain = false;
+
+  void setBundleByDomain(bool shouldBundleByDomain) {
+    setState(() {
+      bundleByDomain = shouldBundleByDomain;
+    });
```

Comment:
instead of using setState directly, I recommend creating a controller class with ValueNotifiers that the UI listens to. If you keep state in the DeepLinkPage state object directly, the state will be lost if the user switches to a separate screen or does something to cause a complete tear down and rebuild of the state object.

We do this for almost every screen in DevTools. Look where `DeepLinkPage` is created to see how other screens pass in a controller that will live for the entire lifecycle of DevTools. Then you can keep anything stateful in the controller object and never have to call set state directly or store data in the widget state.